### PR TITLE
feat: add method to receive User UI language code

### PR DESCRIPTION
**Pull request by shpran**
**Relates to:** _[ADF-167](https://oat-sa.atlassian.net/browse/ADF-167)_

**Changes:** added new helper method to get user language interface code.

---

**Companion PR:** https://github.com/oat-sa/extension-tao-udir/pull/129

### CODE REVIEW

File: helpers/UserHelper.php [hunk 1]
```diff
@@ -153,13 +151,27 @@ public static function getUserName(User $user, $defaultToLabel = false)
...
             $userName = self::getUserLabel($user);
         }
 
         return $userName;
     }
+
+    /**
+     * @param User|string $user
+     *
+     * @return string
```
Comment (boajer) [inline helpers/UserHelper.php:167]:
```suggestion
```

File: helpers/UserHelper.php [hunk 2]
```diff
@@ -153,13 +151,27 @@ public static function getUserName(User $user, $defaultToLabel = false)
...
+     */
+    public static function getUserUILanguageCode($user): string
+    {
+        if (is_string($user)) {
+            $user = self::getUser($user);
+        }
+
```
Comment (gabrielfs7) [inline helpers/UserHelper.php:174]: Do you think we could have strict types as type hint? Mixed type generally opens a precedent to increase coupling and difficult to maintain / refactor the method after.

```suggestion
    public static function getUserUILanguageCode(string $userId): string
    {
        $user = self::getUser($userId);

```

Comment (gabrielfs7) [reply helpers/UserHelper.php:174]: Do you think we should have this type of logic here instead? 

https://github.com/oat-sa/generis/blob/master/common/oatbox/user/UserLanguageService.php

**Review by gabrielfs7 — COMMENTED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Pull request closed without merging**